akka-actor/src/main/scala/akka/util/ImmutableIntMap.scala

```diff
@@ -3,104 +3,111 @@
  */
 package akka.util
 import java.util.Arrays
-
 import akka.annotation.InternalApi
-
 import scala.annotation.tailrec
 
 /**
  * INTERNAL API
  */
 @InternalApi private[akka] object ImmutableIntMap {
-  val empty: ImmutableIntMap =
-    new ImmutableIntMap(Array.emptyIntArray, Array.empty)
-
-  private final val MaxScanLength = 10
+  val empty: ImmutableIntMap = new ImmutableIntMap(Array.emptyLongArray)
 }
 
 /**
  * INTERNAL API
- * Specialized Map for primitive `Int` keys to avoid allocations (boxing).
- * Keys and values are backed by arrays and lookup is performed with binary
- * search. It's intended for rather small (<1000) maps.
+ * Specialized Map for primitive `Int` keys and values to avoid allocations (boxing).
```

Comment:
@patriknw On my machine, the Int-array one seems slightly faster, see: https://github.com/akka/akka/pull/23315/commits/77003e132250eb59da8069ae5dd1b8c8e98cd136